sky/packages/sky/lib/src/rendering/custom_layout.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'box.dart';
+import 'object.dart';
+
+class _MultiChildParentData extends ContainerBoxParentDataMixin<RenderBox> { }
+
+abstract class MultiChildLayoutDelegate {
+  Map<Object, RenderBox> _idToChildMap = new Map<Object, RenderBox>();
```

Comment:
Sure.

Ian suggested using an InheritedWidget to give CustomMultiChildLayout children an Id. So instead of this:
```
new CustomMultiChildLayout(
  childMap: {
    'foo': new Container(height: 100.0, decoration: green),
    'bar': new Container(height: 200.0, decoration: blue)
  }
)
```
You'd write something like this:
```
new CustomMultiChildLayout([
    new Id('foo', child: new Container(height: 100.0, decoration: green)),
    new Id('bar', child: new Container(height: 200.0, decoration: blue))
])
```
As I understand it, this approach obviates calling childMap.keys and childMap.values (and converting them to lists) everytime the CustomMultiChildLayout is built since the ids can be stored in each child's ParentData.